crates/db_views/community_follower_approval/src/impls.rs

```diff
@@ -0,0 +1,412 @@
+use crate::PendingFollowerView;
+use diesel::{
+  dsl::{count, exists, sql},
+  pg::sql_types::Array,
+  select,
+  sql_types::Integer,
+  BoolExpressionMethods,
+  ExpressionMethods,
+  JoinOnDsl,
+  NullableExpressionMethods,
+  QueryDsl,
+};
+use diesel_async::RunQueryDsl;
+use i_love_jesus::SortDirection;
+use lemmy_db_schema::{
+  aliases,
+  newtypes::{CommunityId, InstanceId, PaginationCursor, PersonId},
+  source::{
+    community::{community_actions_keys as key, Community, CommunityActions},
+    person::Person,
+  },
+  traits::PaginationCursorBuilder,
+  utils::{get_conn, limit_fetch, paginate, queries::selects::person1_select, DbPool},
+};
+use lemmy_db_schema_file::{
+  enums::{CommunityFollowerState, CommunityVisibility},
+  schema::{community, community_actions, person},
+};
+use lemmy_utils::error::{LemmyErrorExt, LemmyErrorType, LemmyResult};
+use std::collections::HashMap;
+
+diesel::alias!(community_actions as follower_community_actions: FollowerCommunityActions,
+person as person_instance_check: PersonInstanceCheck,
+community_actions as community_actions_instance_check: CommunityActionInstanceCheck);
+
+impl PendingFollowerView {
+  #[diesel::dsl::auto_type(no_type_alias)]
+  fn joins() -> _ {
+    let follower_community_actions_join = follower_community_actions
+      .on(community::id.eq(follower_community_actions.field(community_actions::community_id)));
+    let follower_id = aliases::person1.field(person::id);
+    let follower_join = aliases::person1.on(
+      follower_community_actions
+        .field(community_actions::person_id)
+        .eq(follower_id)
+        .and(
+          follower_community_actions
+            .field(community_actions::followed_at)
+            .is_not_null(),
+        )
+        .and(community::id.eq(follower_community_actions.field(community_actions::community_id))),
+    );
+
+    community_actions::table
+      .inner_join(community::table)
+      .inner_join(person::table.on(community_actions::person_id.eq(person::id)))
+      .inner_join(follower_community_actions_join)
+      .inner_join(follower_join)
+  }
+
+  pub async fn list_approval_required(
+    pool: &mut DbPool<'_>,
+    mod_id: PersonId,
+    all_communities: bool,
+    unread_only: bool,
+    cursor_data: Option<CommunityActions>,
+    page_back: Option<bool>,
+    limit: Option<i64>,
+  ) -> LemmyResult<Vec<PendingFollowerView>> {
+    let conn = &mut get_conn(pool).await?;
+    let limit = limit_fetch(limit)?;
+
+    let mut query = Self::joins()
+      .filter(community_actions::became_moderator_at.is_not_null())
+      .filter(community::visibility.eq(CommunityVisibility::Private))
+      .select((
+        person1_select(),
+        community::all_columns,
+        follower_community_actions
+          .field(community_actions::follow_state)
+          .nullable(),
+      ))
```

Comment:
You should just be able to use `Self::as_select()`. That's how it is in every other view.